engine_test.go

```diff
@@ -210,3 +210,22 @@ outputs:
 	assert.Equals(t, outputID, "success")
 	assert.Equals(t, outputData.(map[any]any), map[any]any{"message": "Hello, Arca Lot!"})
 }
+
+func Test_ParseSubworkflows(t *testing.T) {
+	fileCache, err := loadfile.NewFileCacheUsingContext(
+		"fixtures/test-subworkflow",
+		map[string]string{
+			"workflow": "test-workflow.yaml",
+		})
+	assert.NoError(t, err)
+	assert.NoError(t, fileCache.LoadContext())
+	outputID, _, outputError, err := createTestEngine(t).RunWorkflow(
+		context.Background(),
+		[]byte(`{}`),
+		fileCache,
```

Comment:
I suppose that verifying the workflow succeeded is a fairly strong test, *but*, to be safe, verifying the key/value pairs you expect in the updated `fileCache` might not be a bad idea?

Reply:
I assert for `success` in each of the main workflow steps.

Reply:
I meant specifically to check the merging of the caches.

Reply:
ooooooooooooooh right! That makes sense.